### PR TITLE
[Release 0.17] Reintroduce fee breakdown tooltips + change price display

### PR DESCRIPTION
**Pull request by W3stside**
Part of swap fee breakdown redesign. 

This is a temporary (temporary part being the tooltips) change until the full changes as discussed here are implemented: https://gnosisinc.slack.com/archives/CPZA1AGKY/p1623942262018500

Prices shown in the INPUT fields have changed:
1. SELL order: 
    a. input field shows the same: the amount user inputs
    b. output field is different: shows the output amount WITHOUT fees
2. BUY order: 
    a. input field is different: shows the required input amount WITHOUT fees
    b. output field shows the same: the amount user exactly wants to buy

Prices show in the auxiliary field labeled `<From | Receive>(incl. fees)` will show the real, fee adjusted amount

#### SELL
image

#### BUY
image

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @W3stside , 
great that Prices show values without without fees and  fee calculation tooltip is back. 

However, I have some doubts about the 2nd tooltip that calculates 'Minimum receive' value. It also contains 'Fee' value (duplicates it from the 1st tooltip),
fee calc
sliip calc

Also, IMHO, it is not intuitively understandable how the Minimum receive value is calculated (`Receive (incl.fee)` - `Slippage Tolerance`) I would suggest to change it to show:
```
Receive (incl. fee)
Slippage Tolerance
Minimum received
```
would suggest


Another case, is that Prices still include fees in the 'Confirm Swap' modals. From my perspective, it would be nice to change then as well: 
SELL order: output field shows the output amount WITHOUT fees
BUY order input field  shows the required input amount WITHOUT fees

then show the next fields under the Price section: 
```
Fee
Receive (incl. fee)
Slippage tolerance
Minimum received
```
confirm swap

What do you think about it?

**Comment by W3stside** [comment]
> What do you think about it?

I generally agree with all the points. I still would like this merged to allow for an RC in 0.17. We can open an issue using your comment to discuss further. wdyt @elena-zh ?

**Comment by elena-zh** [comment]
I would postpone tooltips changing and fix it in another task. 
But I would change confirmation modal in this pr. At least, this: 

- SELL order: output field shows the output amount WITHOUT fees
- BUY order input field shows the required input amount WITHOUT fees

**Review by alfetopito — COMMENTED** [review]
Part from Elena's comments, 2 minor UI issues:

screenshot_2021-07-20_12-50-47

1. Font on the extra field with the price break down seems to be different than the font used on the input. Intended? Or is it just the size different that makes it look different?
2. Approval button is glued to disabled swap button

**Comment by biocom** [comment]
Some observations:
1. Should we add the currency symbol in the tooltip? I think it reads faster if you see:
   `Before fee 34.943 WETH`
   `Fee 7.34 USDT`
2. Agree with @elena-zh to match the confirm modal contents with what we see in the Swap container. IMO Could be post release v.0.17 if it delays things.

**Comment by biocom** [comment]
Approve with comments and this style enhancement PR #996

File: src/custom/components/CurrencyInputPanel/CurrencyInputPanelMod.tsx [hunk 1]
```diff
@@ -147,6 +148,25 @@ export const StyledBalanceMax = styled.button`
     margin-right: 0.5rem;
   `};
 `
+const AuxInformationContainer = styled(Container)`
+  &&&&& {
```
[the remark this answers is not in the document]
Comment (W3stside) [reply src/custom/components/CurrencyInputPanel/CurrencyInputPanelMod.tsx:152]: specificity

**Comment by W3stside** [comment]
> Some observations:
> 
> 1. Should we add the currency symbol in the tooltip? I think it reads faster if you see:
>    `Before fee 34.943 WETH`
>    `Fee 7.34 USDT`
@biocom fees are denoted all in the same currency so imho it's not necessary to add units
> 2. Agree with @elena-zh to match the confirm modal contents with what we see in the Swap container. IMO Could be post release v.0.17 if it delays things.
done in #1007

**Comment by W3stside** [comment]
> > @biocom fees are denoted all in the same currency so imho it's not necessary to add units
> 
> While technically that is the case (fees denoted in the same currency) it's easy to lose context as to what 'the same' currency relates to. I'd say for clarity we should add it. Especially given the tooltip has the primary purpose of making the breakdown clearer and not more abstract.
> 
> E.g. comparing a tooltip proposal from a while back:
> 
> Screen Shot 2021-07-22 at 09 57 02vsScreen Shot 2021-07-22 at 09 59 30
> 
> Btw, notice the dot behind the sum in the last screenshot? That's from this PR.

lol i was so confused when i wrote this but u edited it

yeah ur right, we can add this

**Comment by biocom** [comment]
@W3stside Oh boy, I confused myself right now 😅

**Comment by W3stside** [comment]
@anxolin @alfetopito @biocom please post merge review, this has been on going tho and can better be tested in staging as part of the release candidate

**Pull request merged**